### ISSUE TITLE
checkpoint.pth not generated in v1.6.1 despite checkpointing working per epoch

### ISSUE THREAD

**Issue by kushaldev75** [roboflow/rf-detr#881]
### Search before asking

- [x] #882

### **Bug**

In RF-DETR v1.6.0, there was a known issue where checkpointing at specified epoch intervals and the creation of `checkpoint.pth` (used for resuming training) was not working. This was reported and marked as resolved in v1.6.1.

However, while using **v1.6.1**, I am still encountering a related issue.

**Observed behavior:**

* When setting `checkpoint_interval=1`, model weights are correctly saved for every epoch.
* However, the file `checkpoint.pth` (used to resume training) is **not generated** in the output directory.

**Expected behavior:**

* Along with epoch-wise checkpoints, a `checkpoint.pth` file should be created/updated to allow resuming training.

**Impact:**

* Unable to resume training from the last checkpoint despite checkpointing being enabled.

**Behavior change / regression:**

* In earlier versions (prior to v1.6.0), `checkpoint.pth` used to be saved and updated continuously during training.
* This allowed resuming training even while training got interrupted due to some other issue
* In v1.6.1, this file is no longer being created at all, despite checkpointing being active.

**Additional context:**

* This appears to be a partial fix compared to v1.6.0.
* It is unclear if additional configuration is required for generating `checkpoint.pth`.
* It is also unclear whether `checkpoint.pth` has been deprecated or replaced (e.g., by another file such as `last.ckpt`).

### Environment

- RF-DETR: 1.6.1
- OS: Google Colab (Linux)
- Python: 3.10 (default Colab)
- PyTorch: 2.10.0+cu128 (auto-installed via RF-DETR in Colab)
- CUDA/cuDNN: CUDA enabled (Colab GPU runtime)
- GPU: T4

### Minimal Reproducible Example

```python id="7k2n9x"
# Installation steps in Google Colab
!pip install -q rfdetr==1.6.1 supervision

# After encountering errors, installed with extras
!pip install "rfdetr[train,loggers]"

# Training configuration
from rfdetr import RFDETRLarge

model = RFDETRLarge(num_classes=2)

model.train(dataset_dir=dataset_dir,
            resolution=704,
            epochs=50,
            lr=1e-4,
            batch_size=4,
            grad_accum_steps=4,
            output_dir=output_dir,
            checkpoint_interval=1,
            progress_bar=True)


# During training:
# - Epoch-wise checkpoints are saved
# - checkpoint.pth is not showing up in the output directory
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by kushaldev75** [comment]
Apologies if the "bug" label is misleading. I selected it based on the observed behavior, but this might also be a change in expected functionality or configuration rather than a strict bug.

**Comment by omkar-334** [comment]
@kushaldev75 you can refer to [docs/learn/train/customization.md](https://github.com/roboflow/rf-detr/blob/a77670b/docs/learn/train/customization.md#resume-from-checkpoint)

 **Note:** When `checkpoint_interval=1`, no `last.ckpt` is written. Use `checkpoint_{epoch}.ckpt` (e.g. `output/checkpoint_epoch=4.ckpt`) to resume instead.

**Comment by kushaldev75** [comment]
I see. Thanks for sharing the reference!
